merkle_zeppelin/trees/merkle.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, Callable
+
+from Crypto.Hash import keccak
+from eth_abi import encode
+
+from .binary import BinaryTree
+from .exceptions import MerkleTreeValidationFailed, ValueNotFoundInTree
+
+Leaf = tuple[Any, ...]
+
+
+def keccak256(v: bytes) -> bytes:
+    return keccak.new(data=v, digest_bits=256).digest()
+
+
+class MerkleTree(BinaryTree[bytes]):
+    _raw_leafs_index: dict[Leaf, int]
+    _hashing_function: Callable[[bytes], bytes]
+    _types: list[str]
+
+    def __init__(
+        self,
+        leafs: list[Leaf],
+        types: list[str],
+        hashing_function: Callable[[bytes], bytes] = None,
+    ) -> None:
+        self._hashing_function = hashing_function or keccak256
+        self._types = types
+
+        hash_leaf_pairs = self._get_hash_leaf_pairs(leafs)
+        hash_leaf_pairs.sort(reverse=True)
+
+        ordered_hashed_leafs = [leaf_hash for leaf_hash, _ in hash_leaf_pairs]
+        ordered_leafs = [leaf for _, leaf in hash_leaf_pairs]
+
+        super().__init__(ordered_hashed_leafs)
+
+        self._raw_leafs_index = self._get_raw_leafs_index(ordered_leafs)
+
+    @property
+    def hashed_leafs(self) -> list[bytes]:
+        return self._nodes[self._inner_nodes_number :]
+
+    def validate(self, raise_exception: bool = True) -> bool:
+        for i in range(1, self._inner_nodes_number):
+            left_node_index = self._get_left_children_index(i)
+            right_node_index = self._get_right_children_index(i)
+            calculated_parent = self._calculate_parent_value(
+                self._nodes[left_node_index], self._nodes[right_node_index]
+            )
+
+            if self._nodes[i] != calculated_parent:
+                if not raise_exception:
+                    return False
+
+                raise MerkleTreeValidationFailed()
```

Comment:
https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement
second paragraph ☝️ 

so I think it may stay like that
but anyway thanks for comment!